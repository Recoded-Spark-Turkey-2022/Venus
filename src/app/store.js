// eslint-disable-next-line no-unused-vars
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import listingReducer from '../features/listings/listingSlice';

export const store = configureStore({
  reducer: {
    listing: listingReducer,
  },
  // eslint-disable-next-line no-shadow
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});
