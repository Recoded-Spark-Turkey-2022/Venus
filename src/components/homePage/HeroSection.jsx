import { motion } from 'framer-motion';
import React from 'react';
import logo from '../../assets/homePage-graphic.svg';
import Button from '../UI/Button';

const HeroSection = () => {
  return (
    <section className="min-h-[90vh] flex items-center justify-center">
      <div className=" w-full flex flex-col-reverse md:flex-row  justify-center md:justify-between lg:justify-around ">
        <motion.div
          initial={{ x: '-1000px', scale: 0.5 }}
          animate={{ x: '0px', scale: 1 }}
          transition={{ duration: 0.5 }}
          className="text-center md:text-left"
        >
          <h1 className="text-5xl lg:text-7xl font-bold text-mediumBlue my-5">
            RefuBook
          </h1>
          <h2 className="text-3xl lg:text-5xl text-openRose font-bold mb-4">
            Express Freely
          </h2>
          <p className="text  w-full  md:max-w-[25rem]">
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Impedit
            laboriosam iure dolores illo soluta cum aliquid minima quos culpa
            officia?
          </p>
          <Button>Sign-up</Button>
        </motion.div>
        <motion.div
          initial={{ x: '1000px', scale: 0.5 }}
          animate={{ x: '0px', scale: 1 }}
          transition={{ duration: 0.5 }}
        >
          <img
            className="mx-auto w-[500px] md:w-full md:mx-0"
            src={logo}
            alt="hero-logo"
          />
        </motion.div>
      </div>
    </section>
  );
};

export default HeroSection;
