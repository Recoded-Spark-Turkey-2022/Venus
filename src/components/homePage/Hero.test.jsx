import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import {store} from '../../app/store';
import { BrowserRouter as Router } from "react-router-dom";
import HeroSection from './HeroSection'

jest.mock('react-i18next', () => ({
    useTranslation: () => {
      return {
        t: (str) => str
      };
    },
  }));
  
  it(`renders correctly` , () => {
    const tree = renderer.create(
    <Router>
      <Provider store={store}>
        <HeroSection />
        </Provider>
    </Router>);
    expect(tree).toMatchSnapshot();
});
