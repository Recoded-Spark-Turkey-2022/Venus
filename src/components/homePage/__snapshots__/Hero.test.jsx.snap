// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className=" mt-[70px]  w-full  min-h-[90vh] relative overflow-hidden md:overflow-visible mb-10"
>
  <img
    alt="circle-logo"
    className="absolute image-hero z-0 left-[-500px] md:left-[-200px] lg:left-[-300px] xl:left-[-400px] 2xl:left-[-500px] top-[0px]  w-[100%] h-[90vh]"
    src="signup-vector.svg"
  />
  <div
    className=" items-center flex flex-col-reverse md:flex-row   md:w-full justify-between md:justify-between lg:justify-between "
  >
    <div
      className="text-center md:text-left "
      style={
        Object {
          "transform": "translateX(-1000px) scale(0.5) translateZ(0)",
        }
      }
    >
      <div
        className="z-10 mt-20 md:mt-20  md:mr-10 lg:mr-0"
        data-testid="banner"
      >
        <h1
          className="text-5xl lg:text-7xl font-bold text-mediumBlue my-5"
        >
          RefuBook
        </h1>
        <h2
          className="text-3xl lg:text-5xl text-openRose font-bold mb-4"
        >
          Express Freely
        </h2>
        <p
          className="text  w-full  md:max-w-[25rem]"
        >
          Home.hero
        </p>
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="my-9 px-6 rounded-2xl  py-1"
            id="mediumBlue-button"
            type="button"
          >
            Button.su
          </button>
        </a>
      </div>
    </div>
    <div
      style={
        Object {
          "transform": "translateX(1000px) scale(0.5) translateZ(0)",
        }
      }
    >
      <img
        alt="hero-logo"
        className="mx-auto mt-5 md:mt-0  w-full xl:w-[700px] md:mx-0  md:aspect-square"
        data-testid="hero-img"
        src="homePage-graphic.svg"
      />
    </div>
  </div>
  <div
    className="Toastify"
  />
</section>
`;
